### ISSUE TITLE
How to use Apex Mock Framework for Custom Setting Records?

### ISSUE THREAD

**Issue by VikasPoddar1319** [apex-enterprise-patterns/fflib-apex-mocks#103]
Hello All,

First of all I want to say that this framework is awesome and very fast. We started using it and now converting all test classes using this framework.

But we got stuck wherever we are using getAll method of custom setting. In that case we are unable to mock the custom setting records! Is there any workaround for the same?

Thanks in advance!

**Comment by ImJohnMDaniel** [comment]
G'day @VikasPoddar1319 

To be honest, I have not tried to mock the custom settings records. 

I suppose you could create a selector around the custom setting.  That selector would not necessarily extend fflib_SObjectSelector but it probably could implement fflib_ISObjectSelector.  Then you could have a "getAll()" method that would be a pass through for the true custom setting "getAll()" method.  If you were able to succeed in creating something like that, then you could use ApexMocks to work with the custom setting selector verses trying to mock the custom setting itself.  

@afawcett @daveespo @stohn777 -- any other ideas??

**Comment by stohn777** [comment]
Hi @VikasPoddar1319.

Custom setting records can be created in test classes, so using Mocks is quite possible.  Like @ImJohnMDaniel, I've not recently needed to mock custom settings, although I remember a few years back when he and I were working together, custom settings could not be created in test classes, so we're better off today.

Here's a significant article [1] on the process.  It alludes to the code querying the records directly, which I, personally, would encapsulate in a custom settings class with which I could inject, as opposed to `database.insert`, the test records.

Best wishes.

[1] https://salesforce.stackexchange.com/questions/9565/how-to-get-custom-settings-in-apex-test-code

**Comment by VikasPoddar1319** [comment]
Thanks for the reply @ImJohnMDaniel & @stohn777 

I tried inserting Custom setting records directly & it always worked for me. But now I don't want to do DML operation and use a mocking framework.
As of now, Instead of using the getAll method. I used Select query in my class to get all custom settings & then I mocked that query to return data according to us. So, it worked for me. Hurree

Thanks @stohn777 & @ImJohnMDaniel
